relayer/processor/path_processor.go

```diff
@@ -452,12 +452,37 @@ ChannelHandshakeLoop:
 
 // assembleMsgUpdateClient uses the ChainProvider from both pathEnds to assemble the client update header
 // from the source and then assemble the update client message in the correct format for the destination.
-func (pp *PathProcessor) assembleMsgUpdateClient(src, dst *pathEndRuntime) (provider.RelayerMessage, error) {
-	// If the client state trusted height is not equal to the client trusted state height, we cannot send a MsgUpdateClient
-	// until another block is observed. But we can send the rest without the MsgUpdateClient.
+func (pp *PathProcessor) assembleMsgUpdateClient(ctx context.Context, src, dst *pathEndRuntime) (provider.RelayerMessage, error) {
+	// If the client state height is not equal to the client trusted state height and the client state height is the latest block, we cannot send a MsgUpdateClient
+	// until another block is observed on the counterparty.
+	// If the client state height is in the past, beyond ibcHeadersToCache, then we need to query for it.
 	if !dst.clientTrustedState.ClientState.ConsensusHeight.EQ(dst.clientState.ConsensusHeight) {
-		return nil, fmt.Errorf("observed client trusted height: %d does not equal latest client state height: %d",
-			dst.clientTrustedState.ClientState.ConsensusHeight.RevisionHeight, dst.clientState.ConsensusHeight.RevisionHeight)
+		if int64(dst.clientState.ConsensusHeight.RevisionHeight)-int64(dst.clientTrustedState.ClientState.ConsensusHeight.RevisionHeight) > ibcHeadersToCache {
+			header, err := src.chainProvider.IBCHeaderAtHeight(ctx, int64(dst.clientState.ConsensusHeight.RevisionHeight+1))
+			if err != nil {
+				pp.log.Error("Error getting IBC header at height",
```

Comment:
Since we log and then return the error, this is likely to end up being double-logged. Typically we want to discourage double-logging errors. Probably better to return a wrapped error.

Reply:
done